repos/system_upgrade/common/actors/checkldsoconf/libraries/checkldsoconfiguration.py

```diff
@@ -0,0 +1,133 @@
+import glob
+import os
+
+from leapp import reporting
+from leapp.libraries.common.rpms import has_package
+from leapp.libraries.stdlib import api, CalledProcessError, run
+from leapp.models import InstalledRedHatSignedRPM
+
+LD_SO_CONF_DIR = '/etc/ld.so.conf.d'
+LD_SO_CONF_MAIN = '/etc/ld.so.conf'
+LD_SO_CONF_DEFAULT_INCLUDE = 'ld.so.conf.d/*.conf'
+LD_SO_CONF_COMMENT_PREFIX = '#'
+LD_LIBRARY_PATH_VAR = 'LD_LIBRARY_PATH'
+FMT_LIST_SEPARATOR_1 = '\n- '
+FMT_LIST_SEPARATOR_2 = '\n    - '
+
+
+def _read_file(file_path):
+    with open(file_path, 'r') as fd:
+        return fd.readlines()
+
+
+def _report_custom_ld_so_configuration(summary):
+    reporting.create_report([
+        reporting.Title(
+            'Detected customized configuration for dynamic linker.'
+        ),
+        reporting.Summary(summary),
+        reporting.Remediation(hint=('Remove or revert the custom ld.so configuration and apply the changes '
+                                    'using the ldconfig command.')),
```

Comment:
I am sorry for broken formatting. /o\
IIRC, active SCLs are messing with the envar and in such a case we cannot call it custom configuration. From this point, suggesting that in case of active SCLs they should disable them, could resolve one of detected problems. We do not need to check here if there are some SCLs really if we just say that in such a case it could be source of detected problems.

```suggestion
        reporting.Remediation(hint=('Remove or revert the custom dynamic linker configurations and apply the changes '
                                    'using the ldconfig command. In case of possible active software collections we suggest disabling them persistently.')),
```

Reply:
I see.. added.